### PR TITLE
Added support for masked language modeling (bidirectional models)

### PR DESCRIPTION
**Pull request by shehadak**
This PR builds on the Huggingface subject, which assumes that models are autoregressive (following the `ModelForCausalLM`). This PR adds support for bidirectional models with masked language modeling following the `ModelForMaskedLM`). Since bidirectional models rely on future context, I use a sliding window approach (see https://github.com/google-research/bert/issues/66). In particular, for each text part, up to `w/2` tokens are included for the current part + previous context, and the remaining `w/2` tokens are masked.

The `region_layer_mapping` for the language system was determined by scoring every transformer layer in BERT's encoder against the `Pereira2018.243sentences-linear`, `Pereira2018.384sentences-linear`, and `Blank2014-linear` benchmarks, and choosing the layer with the highest average score.

This PR also provides unit tests for reading time estimation, next word prediction, and neural recording, using the `bert-base-uncased` model. Future models can use the same format, as long as they implement the `ModelForMaskedLM` interface. For example, to add the [base DistilBERT](https://huggingface.co/docs/transformers/model_doc/distilbert) model:

```
model_registry['distilbert-base-uncased'] = lambda: HuggingfaceSubject(model_id='distilbert-base-uncased', region_layer_mapping={
    ArtificialSubject.RecordingTarget.language_system: 'distilbert.transformer.layer.5'}, bidirectional=True)
```

### CODE REVIEW

File: brainscore_language/model_helpers/huggingface.py [hunk 1]
```diff
@@ -115,6 +121,97 @@ def digest_text(self, text: Union[str, List[str]]) -> Dict[str, DataAssembly]:
             if self.neural_recordings:
                 representations = self.output_to_representations(layer_representations, stimuli_coords=stimuli_coords)
                 output['neural'].append(representations)
+        return output
+
+    def _masked_inference(self, text):
```
Comment (mschrimpf) [inline brainscore_language/model_helpers/huggingface.py:126]: there's a lot of duplicate code here (with `_causal_inference`), can we unify this?

File: brainscore_language/models/bert/__init__.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from brainscore_language import model_registry
+from brainscore_language import ArtificialSubject
+from brainscore_language.model_helpers.huggingface import HuggingfaceSubject
+
+model_registry['bert-base-uncased'] = lambda: HuggingfaceSubject(model_id='bert-base-uncased', region_layer_mapping={
+    ArtificialSubject.RecordingTarget.language_system: 'bert.encoder.layer.4'}, bidirectional=True)
```
[the remark this answers is not in the document]
Comment (shehadak) [reply brainscore_language/models/bert/__init__.py:6]: No, the region mapping was determined by scoring each transformer layer in against the three neural benchmarks: Pereira2018.243sentences-linear, Pereira2018.384sentences-linear, and Blank2014-linear benchmarks, and choosing the layer with the highest average score. I'm also happy to use the [neural-nlp scores](https://github.com/mschrimpf/neural-nlp/blob/master/precomputed-scores.csv). Do you have a preference?

File: brainscore_language/model_helpers/huggingface.py (hunk 1 above)
Comment (shehadak) [reply brainscore_language/model_helpers/huggingface.py:105]: I agree. [a4c52f4](https://github.com/brain-score/language/pull/211/commits/a4c52f4f074e16b3520fbe435eea0bb34ee271a1) refactors a lot of the common code into separate methods.

File: brainscore_language/model_helpers/huggingface.py [hunk 3]
```diff
@@ -288,6 +388,8 @@ def _register_hook(self,
         def hook_function(_layer: torch.nn.Module, _input, output: torch.Tensor, key=key):
             # fix for when taking out only the hidden state, this is different from dropout because of residual state
             # see:  https://github.com/huggingface/transformers/blob/c06d55564740ebdaaf866ffbbbabf8843b34df4b/src/transformers/models/gpt2/modeling_gpt2.py#L428
+            if isinstance(output, tuple):
+                output = output[0]
             output = output[0] if len(output) > 1 else output
```
[the remark this answers is not in the document]
Comment (shehadak) [reply brainscore_language/model_helpers/huggingface.py:381]: I see. You're correct, but I think the edge case is when `output` is a single-element tuple, a case which L393 doesn't handle. This can be handled using the following:
```suggestion
            output = output[0] if isinstance(output, tuple) else output
```
What do you think?

File: brainscore_language/models/bert/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mschrimpf) [reply brainscore_language/models/bert/__init__.py:6]: no strong preference, let's just add a comment explaining how we made the choice

File: brainscore_language/model_helpers/huggingface.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mschrimpf) [reply brainscore_language/model_helpers/huggingface.py:381]: 👍

File: brainscore_language/models/bert/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (shehadak) [reply brainscore_language/models/bert/__init__.py:11]: added a comment in [08433e3](https://github.com/brain-score/language/pull/211/commits/08433e3e263fe32c4fbacde05abbab4182d83e05).
